llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.h

```diff
@@ -244,7 +244,9 @@ class AMDGPUInstructionSelector final : public InstructionSelector {
   bool isDSOffset2Legal(Register Base, int64_t Offset0, int64_t Offset1,
                         unsigned Size) const;
   bool isFlatScratchBaseLegal(
-      Register Base, uint64_t FlatVariant = SIInstrFlags::FlatScratch) const;
+      MachineInstr &FullAddr, MachineInstr *PartAddr = nullptr,
```

Comment:
In GlobalISel it is more normal to pass values around as `Register` instead of `MachineInstr&`. Is there a reason you changed this?